hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -313,6 +314,7 @@ public abstract class Dialect implements ConversionContext, TypeContributor, Fun
 
 	private static final Pattern ESCAPE_CLOSING_COMMENT_PATTERN = Pattern.compile( "\\*/" );
 	private static final Pattern ESCAPE_OPENING_COMMENT_PATTERN = Pattern.compile( "/\\*" );
+	private static final Pattern QUERY_PATTERN = Pattern.compile( "^\\s*(select\\b.+?\\bfrom\\b.+?)(\\bwhere\\b.+?)$" );
```

Comment:
Please keep that in mind

You must modify the regular expression.
https://github.com/hibernate/hibernate-orm/pull/9105

https://hibernate.atlassian.net/browse/HHH-18739

Reply:
Thanks for letting me know, I hadn't seen that PR yet. I cannot merge it because reason all the CI builds are failing. Are you sure your PR is from the latest main ?
I could include your fix in my PR, but I would like to give you credit for it, so the other option is that I push my fix first, and then you rebase your PR off the latest main, wdyt?

Reply:
@jrenaat  I will, thank you.